back/controllers/rangeSupervision.js

```diff
@@ -34,13 +39,57 @@ const controller = {
   },
 
   create: async function createSupervision(request, response) {
+    
+    try{
+      //fetches the supervisor id who is assgined to the range.
+      const vastaanottaja = await geetUserEmail(response.req.body.supervisor);
+      //knex part that should be done in models? returns the email based on the id fetched above:
+      async function geetUserEmail(key) {
+        return await knex
+          .from('user')
+          .where({ 'user.id': key })
+          .select('user.email');
+      }
+      //knex part ends.
+      var emailtostring = JSON.stringify(vastaanottaja);
+      //sending fetched email address to mailer.js where it is used to send message that user has been assigned a supervision
+      email('assigned', emailtostring);
+    } catch (error) {
+      console.error(error);
+    }
     return response
       .status(201)
       .send(response.locals.queryResult);
   },
 
   // no return here? may be a cause for a bug
   update: async function updateSupervision(request, response) {
+
+    //updates.range_supervisor returns "absent" if supervisor has not been assigned. it returns "not confirmed" is supervisor is assigned.
+    //if - checks if supervisor is assigned and only sends email if it is set. otherwise it would send email aswell when supervisor is taken off.
+    try{ 
+      console.log(response.locals.updates.range_supervisor);
+      const AbsentChecker = JSON.stringify(response.locals.updates.range_supervisor);
+      const unquoted = String(AbsentChecker.replace(/"([^"]+)":/g, ''));  
+      const NotAbsent= '"not confirmed"';
+      if (unquoted == NotAbsent){
```

Comment:
Avoid using uppercase starting letters: this is a notation reserved for React components. Use camelCase instead:

absentChecker
notAbsent

Reply:
done